### PR TITLE
test: 1203 ensure minimum charge information in prices series

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

Two tests are added. When Charge prices series (D08) are submitted
* no optional charge information is saved
* only minimum charge information is contained in notifications to GridAccesProviders

## References

* #1203

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeriesWithInformationToBeIgnored.xml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{$systemOperatorMarketParticipant}}</cim:chargeTypeOwner_MarketParticipant.mRID>
+                <cim:type>D03</cim:type>
+                <cim:mRID>EA-001</cim:mRID>
+                <cim:name>Should be ignored</cim:name>
+                <cim:description>All ChargeInformation related data should be ignored</cim:description>
+                <cim:priceTimeFrame_Period.resolution>PT1H</cim:priceTimeFrame_Period.resolution>
+                <cim:effectiveDate>{{$isoTimestampPlusOneMonth}}</cim:effectiveDate>
+                <cim:VATPayer>D01</cim:VATPayer>
+                <cim:transparentInvoicing>false</cim:transparentInvoicing>
+                <cim:taxIndicator>false</cim:taxIndicator>
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeriesWithInformationToBeIgnored.xml:40]:
```suggestion
                <cim:taxIndicator>true</cim:taxIndicator>
```

Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeriesWithInformationToBeIgnored.xml:40]: It is on purpose, that I have set the ChargeInformation related properties to the opposite of the value already stored in the Charges database.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs [hunk 2]
```diff
@@ -318,13 +309,66 @@ public async Task Given_ChargeExampleFileWithInvalidBusinessReasonCode_When_Grid
...
+                var actual = GetCharge(chargesAssertDatabaseContext, senderProvidedChargeId, ownerId, chargeType);
+                actual.Should().BeEquivalentTo(expected, x => x.Excluding(y => y!.Points));
+                actual.Points.Should().NotBeEquivalentTo(expected.Points);
+
+                // We need to clear host log after each test is done to ensure that we can assert on function executed on each test run because we only check on function name.
+                Fixture.HostManager.ClearHostLog();
+            }
+
+            [Fact]
+            public async Task Given_ChargePriceMessageWithChargeInformationData_WhenPosted_ThenChargePriceDataAvailableIsNotified()
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs:343]: Can we improve this test name so it states something about the notification contains only mandatory data?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs [hunk 3]
```diff
@@ -318,13 +309,66 @@ public async Task Given_ChargeExampleFileWithInvalidBusinessReasonCode_When_Grid
...
+
+                using var assertionScope = new AssertionScope();
+                var peekResult = await Fixture.MessageHubMock.AssertPeekReceivesRepliesAsync(correlationId, 2);
+                var notification = peekResult.First(s => s.Contains("NotifyPriceList_MarketDocument"));
+                notification.Should().Contain("<cim:process.processType>D08</cim:process.processType>");
+                notification.Should().NotContain("<cim:name>");
+                notification.Should().NotContain("<cim:description>");
+                notification.Should().NotContain("<cim:VATPayer>");
+                notification.Should().NotContain("<cim:transparentInvoicing>");
+                notification.Should().NotContain("<cim:taxIndicator>");
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs:363]: And perhaps expand this with the mandatory data? Charge ID, Owner, Type and effectiveDate"

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs (hunk 2 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs:343]: Yes, sure!

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs (hunk 3 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeIngestionTests.cs:363]: So, with a `notification.Contain()` for each of those?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeriesWithInformationToBeIgnored.xml (hunk 1 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/TestFiles/Charges/PriceSeries/TaxTariffPriceSeriesWithInformationToBeIgnored.xml:40]: It makes sense, keep it 😄

**Review by prtandrup — APPROVED** [review]

**Pull request merged**